pkg/services/ngalert/state/manager.go

```diff
@@ -356,13 +356,15 @@ func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRu
 
 	currentState.LastEvaluationTime = result.EvaluatedAt
 	currentState.EvaluationDuration = result.EvaluationDuration
+	values := NewEvaluationValues(result.Values)
 	currentState.LatestResult = &Evaluation{
 		EvaluationTime:  result.EvaluatedAt,
 		EvaluationState: result.State,
-		Values:          NewEvaluationValues(result.Values),
+		Values:          values,
 		Condition:       alertRule.Condition,
 	}
 	currentState.LastEvaluationString = result.EvaluationString
+	currentState.Values = values
```

Comment:
This is the other half of the fix. We drop the logic in cache, and re-use NewEvaluationValues here.